framework/db/sqlite/QueryBuilder.php

```
@@ -174,15 +174,22 @@ public function resetSequence($tableName, $value = null)
-     * @return string the SQL statement for checking integrity
-     * @throws NotSupportedException this is not supported by SQLite
+     *
+     * @param bool $check Whether to turn on or off the integrity check.
+     * @param string $schema The schema of the tables. Meaningless for SQLite.
+     * @param string $table The table name. Meaningless for SQLite.
+     *
+     * @throws NotSupportedException this is not supported by SQLite.
+     *
+     * @return string The SQL statement for checking integrity.
      */
     public function checkIntegrity($check = true, $schema = '', $table = '')
```

PHPMD (2.15.0)

[error] The method checkIntegrity has a boolean flag argument $check, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)